test/liken_test.exs

```diff
@@ -157,22 +181,81 @@ defmodule LikenTest do
   end
 
   test "a Struct is like the same Struct" do
-    assert %TestStruct{a: 1} |> like?(%TestStruct{a: 1})
+    expected = %TestStruct{a: 1}
+    actual   = %TestStruct{a: 1}
+    assert actual |> must_contain(expected)
   end
 
   test "a Struct is like a Struct with a subset of the attributes" do
-    assert %TestStruct{a: 1, b: 2} |> like?(%TestStruct{a: 1})
+    actual = %TestStruct{a: 1, b: 2}
+    expected = %{a: 1}
+
+    assert actual |> must_contain(TestStruct, expected)
   end
 
   test "a Struct is not like a Struct with a superset of the attributes" do
-    refute %TestStruct{a: 1} |> like?(%TestStruct{a: 1, b: 2})
+    actual = %TestStruct{a: 1}
+    expected = %{a: 1, b: 2}
+
+    func = fn -> assert actual |> must_contain(TestStruct, expected) end
+
+    assert_raise ExUnit.AssertionError, """
+    Actual value was not an exact match for Expected value:
+
+    At:
+    Map[:b]
+
+    Actual:
+    nil
+
+    Expected:
+    2
+
+    """, func
   end
 
   test "a Struct is not like a Struct with the same attributes but different struct types" do
-    refute %TestStruct{a: 1} |> like?(%FooStruct{a: 1})
+    actual = %TestStruct{a: 1}
+    expected = %{a: 1, b: 2}
+
+    func = fn -> assert actual |> must_contain(FooStruct, expected) end
```

Comment:
The type AND the attributes are different here, but the test name implies you'll just be testing the type difference.